### ISSUE TITLE
Reduce privileges to Read only users

### ISSUE THREAD

**Issue by jpinto3488** [umbraco/Umbraco.Cloud.Issues#234]
Right now Read only users on Umbraco Cloud have too many privileges, they are able to set DEBUG ON/OF and to restart an environment, both are WRITE actions and should not be allowed.

Furthermore, if you want to protect an environment with Basic Auth, the only way to allow access to someone is by creating an account with at least READ access, but this will allow the user to clone the project and access the source code.

Sometimes we setup an environment with basic auth to preview features and need to allow access to external users, but we don't want to share the source code with them. We'd need some "guest" users which are allowed ONLY to access the basic authenticated environment, without letting them access the backoffice or the source code. I know we can protect the back office by disabling the user on /umbraco, but they can still get the source code.

I suggest to:
1 - Disable Debug / Restart on READ users
2 - Create a new user level "GUEST" that is allowed only to access basic authentication on protected environments, with no access to source code, not even READ.

**Comment by sitereactor** [comment]
Hey @jpinto3488 thanks for the suggestion! We are planning to revisit / overhaul the permissions and roles in the Umbraco Cloud Portal during August, so will take your suggestion into consideration.
Right now the plan is to better align the roles in the Umbraco Cloud Portal with those of the Umbraco Backoffice, such that it would be possible to invite an editor through the Portal. But a guest user for the Portal sounds like a good idea.

One follow up question: would you want your guest user to have access to the Umbraco backoffice or only the protected website?

**Comment by jpinto3488** [comment]
Hi @sitereactor thanks for the feedback.
I'd need the guest user to have access only to the protected website, often we need to share a protected environment with partners or customers for testing or approval and the only thing we want is to allow them to see the website.
In the past we used IP whitelisting for this and most of the times it did the job. But right now a lot of people is homeworking and IP whitelists are not a solution anymore.

**Comment by sitereactor** [comment]
Alright, makes sense. Thanks!
It sounds like these are people who might not need a user/account if it wasn't for Previewing / Approving the website before going live. So I wonder if it would be a better option to generate a token or a url with a token, which would give them access to the protected website for a set amount of time?

**Comment by jpinto3488** [comment]
Yes this people only needs to preview / approve / test.
I'm not sure if the token would be a solution though. How would that work? You put a token as a querystring on the initial request and then an auth cookie is saved? This may work for website browsing as normal user. But what about API testing?

If we deploy an UmbracoApiController on a protected environment and what to share it with partner developers for testing with their system, the auth token may not be correctly preserved across requests. Or could you just pass the token during all of the requests?

Well, the token is a great idea and may work perfectly most of times. But having a "Guest" user too through basic auth could help for multiple purposes

**Comment by sitereactor** [comment]
Yes, good point about APIs. In that case a guest user / guest login would probably be necessary. But then you'd kinda need double authentication unless that same guest user exists in the backoffice, so you can (re)use it for the Umbraco API access as well, no?

**Comment by jpinto3488** [comment]
Not really. If you build a public UmbracoApiController it doesn't require a backoffice authentication. But you may need to test it on a protected environment before release

**Comment by FransdeJong** [comment]
I'd like to add to this issue.
I totally agree with the need of extra roles in cloud but our usecase is a little different.

We have multiple clients with multiple websites. I now give them portal access without admin rights because its a huge benefit to have all sites in one place.

We would love a editor role on cloud that sort of corresponds with the Umbraco backoffice editor. If I add the client to the project as a editor he/she would be added as editor in the backoffice (seperate issue: https://github.com/umbraco/Umbraco.Cloud.Issues/issues/308) and the following options should be hidden on the project:

![cloud editor](https://user-images.githubusercontent.com/29602162/106617959-582c3700-656f-11eb-9899-aa1051eb1f37.png)

So basically all the developer stuff should be hidden.

This would be a very powerfull tool and a extra USP to sell Cloud to our clients.

Frans

**Comment by jemayn** [comment]
Thank you for reporting this. We're closing this issue as part of a cleanup of older items on the tracker. 

I believe the access rights have changed since this issue was created, and it is now possible to be invited with no "portal" access so users can have solely Umbraco backoffice access.
